### PR TITLE
Handle failure to create cache file directories by throwing exception

### PR DESCRIPTION
**Pull request by dwalluck**
The original version called `new File(configFolder).mkdirs()` without checking for failure. Then it could fail unexpectedly later. I replaced with `Files.createDirectories(path)` which throws an `IOException`. I also removed `fileExists()` and replaced with separate calls to `Files.isRegularFile()` and `Files.isDirectory()` so that we check the actual type as opposed to just the existence.

### CODE REVIEW

**Comment by rnc** [comment]
@thescouser89 Can this be either merged or reviewed please?

File: auth/src/main/java/org/jboss/pnc/bacon/auth/model/CacheFile.java [hunk 1]
```diff
@@ -80,11 +81,13 @@ public static void writeCredentialToCacheFile(
...
     }
 
-    private static void createConfigFolderIfAbsent() {
-        if (!fileExists(Config.getConfigLocation())) {
+    private static void createConfigFolderIfAbsent() throws IOException {
+        Path path = Paths.get(Config.getConfigLocation());
+
+        if (!Files.isDirectory(path)) {
             String configFolder = Config.getConfigLocation();
             log.debug("Creating config folder {}", configFolder);
```
Comment (thescouser89) [inline auth/src/main/java/org/jboss/pnc/bacon/auth/model/CacheFile.java:89]: LGTM! Just a tiny suggestion: I guess we could just print `path.toString()` here and erase line 88? WDYT?

Comment (dwalluck) [reply auth/src/main/java/org/jboss/pnc/bacon/auth/model/CacheFile.java:89]: Yeah, it's basically getting `Config.getConfigLocation()` twice, once as a `Path` and then again as a `String`.

File: auth/src/main/java/org/jboss/pnc/bacon/auth/model/CacheFile.java [hunk 2]
```diff
@@ -39,29 +39,29 @@ public static void writeCredentialToCacheFile(
...
         try {
+            createConfigFolderIfAbsent();
+
             Map<String, Credential> data = Collections
                     .singletonMap(generateUsernameMd5(keycloakUrl, realm, username), credential);
             CacheFile cacheFile = new CacheFile();
             cacheFile.setCachedData(data);
             mapper.writeValue(file, cacheFile);
-            setOwnerFilePermissions(getCacheFile());
+            setOwnerFilePermissions(file.toPath());
```
Comment (dwalluck) [inline auth/src/main/java/org/jboss/pnc/bacon/auth/model/CacheFile.java:54]: I also changed this to take a `Path` so as to not call `getCacheFile()` a second time.

**Pull request merged**